### ISSUE TITLE
.or with like not working.. 

### ISSUE THREAD

**Issue by JulianSwales** [supabase-community/supabase-flutter#207]
**Describe the bug**

    print('Search Values: email: $email phone: $phone phone1: $phone1');
    final response = await supabase
        .from(tableName())
        .select()
        .or('email.like.%$email%,phone.like.%$phone%,phone.like.%$phone1%')
        .execute();
 
With the following values the search works:
 Search Values: email: 306 phone: +1 (306 phone1: 306

Query in supabase works: 
select * from profiles where email like '%306%' or phone like '%306%' or phone like '%+1 (306%'

With the following values the search does NOT work:
 Search Values: email: 3062 phone: +1 (306) 2 phone1: 3062

Query in supabase works: 
select * from profiles where email like '%3062%' or phone like '%3062%' or phone like '%+1 (306) 2%'

**To Reproduce**

**Expected behavior**
I expect a return value for both queries based on the data and that the query works in supabase

**Screenshots**

**Smartphone (please complete the following information):**
 - Version [3.3]

**Additional context**
I may be doing something wrong, since this is my first attempt at using .or in the solution..

**Comment by dshukertjr** [comment]
@JulianSwales Thanks for opening an issue.

I think the `()` are the ones messing things up here. This definitely needs to be better documented, but there are some [reserved characters](https://postgrest.org/en/stable/api.html?highlight=reserved#reserved-characters) that you have to wrap with `""`. 

In your case, could you try the following?

```dart
final response = await supabase
        .from(tableName())
        .select()
        .or('email.like.%$email%,phone.like."%$phone%",phone.like.%$phone1%')
        .execute();
```

**Comment by JulianSwales** [comment]
@dshukertjr  Yes that was the issue the "" around that one variable have the .or working now.. I had seen the reserved characters comment, but ignored it because I was using variables in the .or string, forgetting that they would be expanded and have the issue. Thanks for your help.
